tests/checkings.test.js

```diff
@@ -1,96 +1,75 @@
-    await CheckingPage.checkingReset()
-    await expect(CheckingPage.checkingNameInput).toHaveText('')
+    await HomePage.accessViewCheckingTable()
+    await CheckingPageTable.changeCheckingTransactionsTableNext()
+    await expect(CheckingPageTable.checkingTableTextInfo).toHaveText('Showing 11 to 20 of 24 entries')
+    await CheckingPageTable.changeCheckingTransactionsTableNext()
+    await expect(CheckingPageTable.checkingTableTextInfo).toHaveText('Showing 21 to 24 of 24 entries')
+    await CheckingPageTable.changeCheckingTransactionsTablePrevious()
+    await expect(CheckingPageTable.checkingTableTextInfo).toHaveText('Showing 11 to 20 of 24 entries')
+    await CheckingPageTable.changeCheckingTransactionsTablePrevious()
+    await expect(CheckingPageTable.checkingTableTextInfo).toHaveText('Showing 1 to 10 of 24 entries')
 })
-it('Should create a Standard Individual Checking', async ()=> {
-    await HomePage.displayCheckingMenu() 
-    await HomePage.accessNewCheckingForm()
-    await CheckingPage.chooseStandardChecking()
-    await CheckingPage.chooseIndividualChecking()
-    await CheckingPage.sendText(CheckingPage.checkingNameInput, 'Standard Individual Checking')
-    await CheckingPage.sendText(CheckingPage.checkingDepositInput, '25')
-    await CheckingPage.checkingSubmit()
-    await expect(CheckingPage.newCheckingConfirmation).toBeDisplayed()
-})
-it('Should switch checking', async ()=> {
-    await CheckingPage.activateSwitchChecking()
-    await expect(CheckingPage.checkingBody).toBeDisplayed()
-})
-it('Should create a Standard Joint Checking', async ()=> {
-    await HomePage.displayCheckingMenu() 
-    await HomePage.accessNewCheckingForm()
-    await CheckingPage.chooseStandardChecking()
-    await CheckingPage.chooseJointChecking()
-    await CheckingPage.sendText(CheckingPage.checkingNameInput, 'Standard Joint Checking')
-    await CheckingPage.sendText(CheckingPage.checkingDepositInput, '50.47')
-    await CheckingPage.checkingSubmit()
-    await expect(CheckingPage.newCheckingConfirmation).toBeDisplayed()
-})
-it('Should create a Interest Individual Checking', async ()=> {
+it('Should filter the table with the word "Income"', async ()=> {
     await HomePage.displayCheckingMenu() 
-    await HomePage.accessNewCheckingForm()
-    await CheckingPage.chooseInterestChecking()
-    await CheckingPage.chooseIndividualChecking()
-    await CheckingPage.sendText(CheckingPage.checkingNameInput, 'Interest Individual Checking')
-    await CheckingPage.sendText(CheckingPage.checkingDepositInput, '1000000')
-    await CheckingPage.checkingSubmit()
-    await expect(CheckingPage.newCheckingConfirmation).toBeDisplayed()
+    await HomePage.accessViewCheckingTable()
+    await CheckingPageTable.sendText(CheckingPageTable.checkingSearchInput, 'Income')
+    const rows = await $$('#transactionTable tr');
+    const columns = await rows[1,2,3,4].$$('td');
+    await expect(columns[1]).toHaveText('Income'); 
```

Comment:
Remember about keeping abstraction layers separated, so no direct selectors should be used at test layer, but always interacted through page objects. Trun this into methods on the corresponding page object and implement this assertion considering that structure.

Also, the general approach you are using is basically correct (reading rows as an array of rows, and then checking columns looking for instances of the "Income" text), but there are flaws in the implementation. Once you have the rows, you need to extract a specific column from each of them, one at a time. `rows[1,2,3,4]` is not valid syntax here, and this will break if you run it (make sure to run tests after changes to make sure they keep working).

One simple approach for solving this would be using a parameterized css selector, that returns you the relevant column from the provided row, such as: 

```javascript

`#transactionTable>tbody>tr:nth-child(${row})>td:nth-child(4)`

```

Where `row` in this case is passed as parameter to the method that takes care of extracting this. This selector will return the element representing the "cell" belonging to the 4th column from the nth row (that row number being controlled by the parameter).

After having that, you can do (in the test), something like:

```javascript

let tableLength = await CheckingPage.getTableNumberOfRows();
let cell;
for(let i=0; i< rowsLength; i++){
     cell = await CheckingPage.getCellFromForRow(i);
     await expect(cell).toHaveTextContaining("Income");
}
```

Or something like that. Notice you'll have to implement a couple of auxiliary methods for this

Reply:
Row, is an array of the table length? How do I get the number of rows in the DOM?

Reply:
`row` in the example selector I provided is an index, a numerical value representing the order of the row you want to retrieve.

For the second action, you can locate all rows from the transactions table into an array (using `$$('tr')` syntax) and then return the array's length, or something like that